Tests/PerformanceTest/XCTestCase+InstructionsCount.swift

```diff
@@ -0,0 +1,74 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import XCTest
+import _InstructionCounter
+
+extension XCTestCase {
+  private var baselineURL: URL {
+    if ProcessInfo.processInfo.environment["SWIFTCI_USE_LOCAL_DEPS"] == "1" {
+      return URL(fileURLWithPath: #file)
+        .deletingLastPathComponent()
+        .appendingPathComponent("ci-baselines.json")
+    } else {
+      return URL(fileURLWithPath: #file)
+        .deletingLastPathComponent()
+        .appendingPathComponent("baselines.json")
+    }
+  }
+
+  func measureInstructions(_ baselineName: StaticString = #function, block: () -> Void, file: StaticString = #file, line: UInt = #line) throws {
+    let strippedBaselineName = "\(baselineName)".replacingOccurrences(of: "()", with: "")
+    var baseline: UInt64? = try baseline(for: strippedBaselineName)
+
+    if let environmentBaseline = ProcessInfo.processInfo.environment[strippedBaselineName] {
+      baseline = UInt64(environmentBaseline)
+    }
+
+    guard let baseline = baseline else {
+      XCTFail("Missing baseline for \(strippedBaselineName)", file: file, line: line)
+      return
+    }
+
+    let startInstructions = getInstructionsExecuted()
+    block()
+    let endInstructions = getInstructionsExecuted()
+    let numberOfInstructions = endInstructions - startInstructions
+
+    let lowerBaseline = UInt64(Double(baseline) * 0.97)
+    let upperBaseline = UInt64(Double(baseline) * 1.03)
+    let baselineRange = lowerBaseline...upperBaseline
+
+    XCTAssertTrue(
+      baselineRange.contains(numberOfInstructions),
+      """
+      Number of instructions '\(numberOfInstructions)'
+      is not within range \(lowerBaseline)-\(upperBaseline)
+      """,
+      file: file,
+      line: line
+    )
+  }
+
+  private func baseline(for key: String) throws -> UInt64? {
+    let baselineData = try? Data(contentsOf: baselineURL)
+
+    guard let data = baselineData else {
+      fatalError("No baseline.json provided")
```

Comment:
I’d prefer to `throw` or `return nil` here so we don’t take down the entire test execution if the baseline file is missing.